test/load-generator/boulder-calls.go

```diff
@@ -582,3 +584,75 @@ func postAsGet(s *State, ctx *context, url string, latencyTag string) (*http.Res
 
 	return s.post(url, requestPayload, ctx.ns, latencyTag, http.StatusOK)
 }
+
+func popCertificate(ctx *context) string {
+	certIndex := mrand.Intn(len(ctx.certs))
+	certURL := ctx.certs[certIndex]
+	ctx.certs = append(ctx.certs[:certIndex], ctx.certs[certIndex+1:]...)
+	return certURL
+}
+
+func getCert(s *State, ctx *context, url string) ([]byte, error) {
+	latencyTag := "/acme/cert/{serial}"
+	resp, err := postAsGet(s, ctx, url, latencyTag)
+	if err != nil {
+		return nil, fmt.Errorf("%s bad response: %s", url, err)
+	}
+	defer resp.Body.Close()
+	return ioutil.ReadAll(resp.Body)
+}
+
+// revokeCertificate removes a certificate url from the context, retrieves it,
+// and sends a revocation request for the certificate to the ACME server.
+// The revocation request is signed with the account key rather than the certificate
+// key.
+func revokeCertificate(s *State, ctx *context) error {
+	if len(ctx.certs) < 1 {
+		return errors.New("No certificates in the context that can be revoked")
+	}
+
+	certURL := popCertificate(ctx)
```

Comment:
It will be a divergence from the other load generator actions but I think it might make sense to extend this `revokeCertificate` action to have a configurable probability of being skipped so we can revoke some certs during the run, but not all of them.

Maybe another config param at the level of `"challengeStrategy"` like `"revokeChance"` that's a float between `0.0` and `1.0` indicating how often an issued certificate is revoked when the `revokeCertificate` action pops a cert from the context? I could imagine eventually wanting to add `"deactivateAuthorization"` or `"keyChange"` actions that might also benefit from some chance.

What do you think?

Reply:
I think that makes sense, I'll have a quick stab at it here.